### PR TITLE
React/week3

### PR DESCRIPTION
**Pull request by JyothiNandyala**

### CODE REVIEW

File: react/week3/fetchtodos/src/App.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  height: 30px;
+  font-size: large;
+  background-color: #64fdca;
+  margin-left: 20px;
+}
+button:hover {
+  background-color: chartreuse;
+  border: 2px solid blue;
+}
+.addbutton {
```
Comment (obelitz) [inline react/week3/fetchtodos/src/App.css:27]: Your class names vary a bit in style :) 
I'd consider changing this to `.add-button` to follow the other ones

File: react/week3/fetchtodos/src/FetchTodos.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    >
+      {props.children}
+    </div>
+  );
+};
+
+const API_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+function TodoList({ id, description, deadline, removeTodo }) {
```
Comment (obelitz) [inline react/week3/fetchtodos/src/FetchTodos.js:23]: This component seems to be a `TodoItem` and the one you've named `RenderTodos` seems to be the actual `TodoList`.

If you want to make it easier to read through, I'd split the `BorderWrapper`, `TodoItem`, and `TodoList` components into each their file :)

File: react/week3/fetchtodos/src/FetchTodos.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    if (dd < 10) {
+      dd = "0" + dd;
+    }
+    if (mm < 10) {
+      mm = "0" + mm;
+    }
+
+    today = yyyy + "-" + mm + "-" + dd;
+    return today;
+  };
```
Comment (obelitz) [inline react/week3/fetchtodos/src/FetchTodos.js:88]: This function does not need to be local to the `RenderTodos`-component. 
If you want to re-use it in the future, you can place it outside the component

File: react/week3/fetchtodos/src/FetchTodos.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        </div>
+      </TodoComponent>
+    </>
+  );
+}
+//This component is for input elements
+function RenderTodos() {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [apidata, setApiData] = useState([]);
```
Comment (obelitz) [inline react/week3/fetchtodos/src/FetchTodos.js:72]: I'd call this something like `apiTodos` - or simply `todos`. 
That makes it easy to see, what it contains

File: react/week3/fetchtodos/src/FetchTodos.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          <label htmlFor="tododescription">Todo Description</label>
+          <input
+            type="text"
+            placeholder="description"
+            value={description}
+            onChange={(event) => {
+              const value = event.target.value;
+              setDescription(value);
+            }}
+          />
```
Comment (obelitz) [inline react/week3/fetchtodos/src/FetchTodos.js:134]: Nice with the labels!
If you want them to be more accessible, you need to add the same value in the `htmlFor` and in the `id`:
```
<label htmlFor="tododescription">Todo Description</label>
<input
  type="text"
  id="tododescription"
  ...
/>
```

File: react/week3/fetchtodos/src/Timer.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useState } from "react";
+
+function Timer() {
+  const [time, setTime] = useState(0);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTime(time + 1);
+    }, 1000);
+    return () => clearTimeout(timer);
```
Comment (obelitz) [inline react/week3/fetchtodos/src/Timer.js:10]: Nice with the cleanup

**Review by obelitz — COMMENTED** [review]
Nice job!
I'd just move the components into their own files for readability

File: react/week3/fetchtodos/src/Timer.js (hunk 6 above)
Comment (JyothiNandyala) [reply react/week3/fetchtodos/src/Timer.js:10]: Thank you obelitz

File: react/week3/fetchtodos/src/FetchTodos.js (hunk 5 above)
Comment (JyothiNandyala) [reply react/week3/fetchtodos/src/FetchTodos.js:134]: ya sure obelitz i will do that ,thank you

File: react/week3/fetchtodos/src/FetchTodos.js (hunk 4 above)
Comment (JyothiNandyala) [reply react/week3/fetchtodos/src/FetchTodos.js:72]: i will change what you said obelitz

**Pull request merged**